dialogflow/api/DialogflowSamples/DetectIntentStream.cs

```diff
@@ -0,0 +1,122 @@
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+using CommandLine;
+using Google.Cloud.Dialogflow.V2;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace GoogleCloudSamples
+{
+    public class DetectIntentStream
+    {
+        public static void RegisterCommands(VerbMap<object> verbMap)
+        {
+            verbMap
+                .Add((DetectIntentFromStreamOptions opts) =>
+                     DetectIntentFromStreamAsync(opts.ProjectId, opts.SessionId, opts.FilePath).Result);
+        }
+
+        [Verb("detect-intent:streams", HelpText = "Detect intent from stream")]
+        public class DetectIntentFromStreamOptions : OptionsWithProjectIdAndSessionId
+        {
+            [Value(0, MetaName = "file", HelpText = "Path to the audio file", Required = true)]
+            public string FilePath { get; set; }
+        }
+
+        // [START dialogflow_detect_intent_streaming]
+        public static async Task<object> DetectIntentFromStreamAsync(
+            string projectId,
+            string sessionId,
+            string filePath)
+        {
+            var sessionsClient = SessionsClient.Create();
+            var sessionName = SessionName.Format(projectId, sessionId);
+
+            // Initialize streaming call, retrieving the stream object
+            var streamingDetectIntent = sessionsClient.StreamingDetectIntent();
+
+            // Define a task to process results from the API
+            var responseHandlerTask = Task.Run(async () =>
+            {
+                var responseStream = streamingDetectIntent.ResponseStream;
+                while (await responseStream.MoveNext())
+                {
+                    var response = responseStream.Current;
+                    var queryResult = response.QueryResult;
+
+                    if (queryResult != null)
+                    {
+                        Console.WriteLine($"Query text: {queryResult.QueryText}");
+                        if (queryResult.Intent != null)
+                        {
+                            Console.Write("Intent detected:");
+                            Console.WriteLine(queryResult.Intent.DisplayName);
+                        }
+                    }
+                }
+            });
+
+            // Instructs the speech recognizer how to process the audio content.
+            // Note: hard coding audioEncoding, sampleRateHertz for simplicity.
+            // Audio encoding of the audio content sent in the query request.
```

Comment:
nit: I believe this line is out of place now.

Reply:
I believe that the comment correctly describes lines 76-84. The QueryInput object includes the configuration files that the speech recognizer will use when recognizing the audio data. The configuration hard codes several settings that developers will need to change for their own data.

Let me know if I'm missing something.

Reply:
I meant just the last line, that seems better on top of line 80. But, yeah, it's really not bad if here, leave it as is.

Reply:
Fair enough, I think that line is probably extraneous. Done.